src/components/PrivateRoute.tsx

```diff
@@ -18,10 +18,14 @@ export const PrivateRoute: React.FC<{
     const [loading, setLoading] = useState(true);
 
     const [authed, setAuthed] = useState<boolean | null>();
-    app.auth().onAuthStateChanged(function (user) {
-        setAuthed(Boolean(user));
-        setLoading(authed == null);
-    });
+    useEffect(() => {
+        const unSubscribe = app.auth().onAuthStateChanged(function (user) {
+            setAuthed(Boolean(user));
+            setLoading(authed == null);
+        });
+        console.log('Authenticated');
+        return unSubscribe;
```

Comment:
Does `unSubscribe` need to be invoked in a return anonymous function or will it automatically be called on return?

Reply:
Both work because `unSubscribe `is an anonymous function itself.

Reply:
but I'll wrap it in a function incase someone unsubs from something else in the future